### PR TITLE
[TKET-1613] Remove Rebase<Target> methods to simplify API.

### PR DESCRIPTION
**Pull request by Roland-djee**
Fixes [TKET-1613](https://cqc.atlassian.net/browse/TKET-1613?atlOrigin=eyJpIjoiMWU4NWJmMTZhNWM0NGNkOWEyNTIzYThkNzViNjA5MmYiLCJwIjoiaiJ9).

This PR addresses the removal of now obsolete `Rebase<Target>` methods in the Python API.

### CODE REVIEW

File: pytket/binders/passes.cpp [hunk 1]
```diff
@@ -443,29 +426,6 @@ PYBIND11_MODULE(passes, m) {
 
   /* Pass generators */
 
-  m.def(
-      "RebaseCustom", &gen_rebase_pass,
```
Comment (cqc-alec) [inline pytket/binders/passes.cpp:447]: Do not remove this one!

File: pytket/binders/passes.cpp [hunk 2]
```diff
@@ -360,23 +360,6 @@ PYBIND11_MODULE(passes, m) {
...
-      "RebaseHQS", &RebaseHQS, "Converts all gates to ZZMax, PhasedX and Rz.");
-  m.def(
-      "RebaseProjectQ", &RebaseProjectQ,
-      "Converts all gates to SWAP, CRz, CX, CZ, H, X, Y, Z, S, T, V, Rx, "
-      "Ry and Rz.");
-  m.def(
-      "RebasePyZX", &RebasePyZX,
-      "Converts all gates to SWAP, CX, CZ, H, X, Z, S, T, Rx and Rz.");
-  m.def("RebaseQuil", &RebaseQuil, "Converts all gates to CZ, Rx and Rz.");
-  m.def("RebaseTket", &RebaseTket, "Converts all gates to CX and TK1.");
```
Comment (cqc-alec) [inline pytket/binders/passes.cpp:374]: Do not remove this one!

**Comment by cqc-alec** [comment]
The idea is to remove backend-specific rebase passes -- _not_ `RebaseTket` or `RebaseCustom` (which are used extensively).

**Comment by Roland-djee** [comment]
> The idea is to remove backend-specific rebase passes -- _not_ `RebaseTket` or `RebaseCustom` (which are used extensively).

OK got that.

**Comment by Roland-djee** [comment]
@ss2165 suggested to use `RebaseCustom` in place of `auto_rebase_pass`.

**Comment by cqc-alec** [comment]
Could you please also remove `SquashHQS` (although not a rebase pass it makes sense to do it at the same time)?

File: pytket/binders/passes.cpp [hunk 3]
```diff
@@ -360,23 +360,7 @@ PYBIND11_MODULE(passes, m) {
       "gates."
       "\n\n:param allow_swaps: whether to allow implicit wire swaps",
       py::arg("allow_swaps") = true);
-  m.def("RebaseCirq", &RebaseCirq, "Converts all gates to CZ, PhasedX and Rz.");
```
[the remark this answers is not in the document]
Comment (Roland-djee) [reply pytket/binders/passes.cpp:363]: Done.

**Comment by Roland-djee** [comment]
> Could you please also remove `SquashHQS` (although not a rebase pass it makes sense to do it at the same time)?

In eb37a11.

File: pytket/binders/passes.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (cqc-alec) [reply pytket/binders/passes.cpp:367]: Please make the change in this PR.

[the remark this answers is not in the document]
Comment (cqc-alec) [reply pytket/binders/passes.cpp:367]: PS. Please include in the changelog a list of the API functions  removed.

File: pytket/tests/predicates_test.py [hunk 4]
```diff
@@ -602,11 +587,6 @@ def test_library_pass_config() -> None:
     assert SynthesiseOQC().to_dict()["StandardPass"]["name"] == "SynthesiseOQC"
     assert SynthesiseUMD().to_dict()["StandardPass"]["name"] == "SynthesiseUMD"
     # Share name with SquashCustom
```
Comment (cqc-alec) [inline pytket/tests/predicates_test.py:589]: This comment doesn't make sense any more.

**Review by cqc-alec — CHANGES_REQUESTED** [review]
Just a minor thing.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**